### DISCUSSION TITLE
Create a Security Policy

### DISCUSSION THREAD

**Discussion by joycebrum** [ReactiveX/rxjs#7265] [Report issues other than bug]
A Security Policy is a GitHub standard document (`SECURITY.md`) that can be seen in the "Security Tab"  and in the About section of the homepage to instruct users about how to report vulnerability in the safest and most efficient way possible.

Security Tab:
https://github.com/joycebrum/rxjs/security

About section:
<img width="322" alt="image" src="https://github.com/ReactiveX/rxjs/assets/22223372/4dc6e185-11be-422d-8d49-3eae2a20dc77">

It is both a [Scorecard Recommendation](https://github.com/ossf/scorecard/blob/main/docs/checks.md#security-policy) (being a security measure of medium priority) and a [Github Recommendation](https://docs.github.com/en/code-security/getting-started/securing-your-repository#setting-a-security-policy) to have a Security Policy configured to the repo.

There are a few ways to receive such disclosures:
- have an email or website available to receive such reports; and/or
- use [GitHub's private vulnerability reporting](https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository) feature.

If you're interested in GitHub's feature, it must be [activated for the repository](https://docs.github.com/en/code-security/security-advisories/repository-security-advisories/configuring-private-vulnerability-reporting-for-a-repository):
1. Open the repo's settings
2. Click on [Code security & analysis](https://github.com/ReactiveX/rxjs/settings/security_analysis)
3. Click "Enable" for "Private vulnerability reporting"

I can send a PR with a draft policy (such as https://github.com/joycebrum/rxjs/security), just let me know.

**Comment by benlesh** [comment]
This is interesting. I'm trying to imagine what sort of security vulnerabilities RxJS could have though. Particularly in the latest version we don't have any dependencies, and we don't generally deal with network code or file system things.

Please do send a PR for this, it doesn't seem like it could cause any harm... Although I do worry about our ability to respond in a timely manner, given this is all volunteer work. It's still good to have a secure notification channel I guess.

**Reply by voliva** [reply]
A possible security vulnerability could be that it introduces a vector for a denial-of-service attack.

**Comment by joycebrum** [comment]
Created a PR https://github.com/ReactiveX/rxjs/pull/7273
